From 96404391f8235457124f9ac7072f6249e05ded08 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 31 Aug 2015 18:24:55 +0800
Subject: [PATCH 0352/5242] MLK-10072 ARM: dts: imx6sx: add uart2/5 support
 for imx6sx-sabreauto board

commit  ea357c9bd4f5f1ff8281676340831d90fc1b1bad from
https://source.codeaurora.org/external/imx/linux-imx.git

Add uart2 and uart5 support for imx6sx-sabreauto board.

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sabreauto.dts |   41 ++++++++++++++++++++++++++++++++
 1 file changed, 41 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx-sabreauto.dts b/arch/arm/boot/dts/imx6sx-sabreauto.dts
index d634a1f..77770c1 100644
--- a/arch/arm/boot/dts/imx6sx-sabreauto.dts
+++ b/arch/arm/boot/dts/imx6sx-sabreauto.dts
@@ -151,6 +151,22 @@
 	status = "okay";
 };
 
+&uart2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart2_1>;
+	status = "okay";
+};
+
+&uart5 { /* for bluetooth */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart5_1>;
+	fsl,uart-has-rtscts;
+	status = "okay";
+	/* for DTE mode, add below change */
+	/* fsl,dte-mode;*/
+	/* pinctrl-0 = <&pinctrl_uart5dte_1>; */
+};
+
 &usbotg1 {
 	vbus-supply = <&reg_usb_otg1_vbus>;
 	pinctrl-names = "default";
@@ -305,6 +321,31 @@
 		>;
 	};
 
+	pinctrl_uart2_1: uart2grp-1 {
+		fsl,pins = <
+			MX6SX_PAD_GPIO1_IO07__UART2_RX 0x1b0b1
+			MX6SX_PAD_GPIO1_IO06__UART2_TX 0x1b0b1
+		>;
+	};
+
+	pinctrl_uart5_1: uart5grp-1 {
+		fsl,pins = <
+			MX6SX_PAD_KEY_ROW3__UART5_RX    0x1b0b1
+			MX6SX_PAD_KEY_COL3__UART5_TX    0x1b0b1
+			MX6SX_PAD_KEY_ROW2__UART5_CTS_B 0x1b0b1
+			MX6SX_PAD_KEY_COL2__UART5_RTS_B 0x1b0b1
+		>;
+	};
+
+	pinctrl_uart5dte_1: uart5dtegrp-1 {
+		fsl,pins = <
+			MX6SX_PAD_KEY_ROW3__UART5_TX	0x1b0b1
+			MX6SX_PAD_KEY_COL3__UART5_RX	0x1b0b1
+			MX6SX_PAD_KEY_ROW2__UART5_RTS_B	0x1b0b1
+			MX6SX_PAD_KEY_COL2__UART5_CTS_B	0x1b0b1
+		>;
+	};
+
 	pinctrl_usb_otg1_vbus: usbotg1vbusgrp {
 		fsl,pins = <
 			MX6SX_PAD_GPIO1_IO09__GPIO1_IO_9     0x10b0
-- 
1.7.9.5

